### DISCUSSION TITLE
Lockfile mismatch in GitHub workflow

### DISCUSSION THREAD

**Discussion by baymac** [pnpm/pnpm#6633] [General]
`pnpm install` is failling in github workflow. There is no issue when installing locally. Here's error message GitHub workflow throws on install:

```
ERR_PNPM_LOCKFILE_CONFIG_MISMATCH  Cannot proceed with the frozen installation. The current "settings.autoInstallPeers" configuration doesn't match the value found in the lockfile
```

With cmd `pnpm install --no-frozen-file` the install passes. What am I doing wrong here?

**Comment by azzamsa** [comment]
Same here, I run `pnpm install --no-frozen-lockfile` but the lockfile doesn't change.

**Reply by baymac** [reply]
For me this error was due to mismatch of `pnpm` version in local and in CI. Once I pinned the `pnpm` version in CI, it worked fine for me.

**Reply by azzamsa** [reply]
I use v8 locally, and v7 in CI. Because when I set v8 in CI, the  CI breaks. https://github.com/pnpm/pnpm/issues/6307#issuecomment-1488929548

I think I will use v7 both locally and CI for now

**Comment by YU000jp** [comment]
Try this

`pnpm-lock.yaml`
```
settings:
  autoInstallPeers: false
```
Change: true → false

**Reply by Shuunen** [reply]
not a good idea because it will be overwritten

**Comment by fourdim** [comment]
Actually, you need to bump your pnpm version to 8 in the github action file.

**Reply by Shuunen** [reply]
Yes this was the real issue for me, thanks @fourdim 👍

**Reply by thejasonxie** [reply]
or `version: latest` works as well.

**Reply by fabriziodevincenzi** [reply]
How? I think I'm doing something wrong

**Reply by thejasonxie** [reply]
https://pnpm.io/continuous-integration#github-actions

it’s the version:8 under pnpm setup

**Comment by manuelernestog** [comment]
You can add .npmrc file in the root of the project with the config

```
auto-install-peers=false
``` 
and this should works for local and GitHub workflow

**Comment by imleeou** [comment]
I modified the version of pnpm in the ` .github/workflows/deploy.yml` file, changing 7 to latest, which resolved the issue.
```yml
  - uses: pnpm/action-setup@v2
        with:
          # version: 7
          version: latest
```
